sphinx_gallery/gen_gallery.py

```diff
@@ -76,6 +78,7 @@ def __call__(self, gallery_conf, script_vars):
     'download_all_examples': True,
     'abort_on_example_error': False,
     'only_warn_on_example_error': False,
+    'recommend_n_examples': True,
```

Comment:
I think that it will be more readable/understandable to have a nested dictionary structure (as for `log_level`). It will also allow more options if you enable different ways of constructing the similarity matrix (word counts vs. backrefs).

```suggestion
    'recommender': {'enable': True, 'n_examples': 5},
```